### PR TITLE
Purge GHC 8.8

### PR DESCRIPTION
**Pull request by michaelpj**
We're also dropping support for this.

Fairly mechanical, mostly just dealing with macros and removing references.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/3287"></a>

### CODE REVIEW

**Review by uhbif19 — APPROVED** [review]

File: ghcide/src/Development/IDE/GHC/Compat/Core.hs [hunk 1]
```diff
@@ -943,11 +889,7 @@ unrestricted = id
 
 mkVisFunTys :: [Scaled Type] -> Type -> Type
 mkVisFunTys =
-#if __GLASGOW_HASKELL__ == 808
-  mkFunTys
-#else
   TcType.mkVisFunTys
```
Comment (andys8) [inline ghcide/src/Development/IDE/GHC/Compat/Core.hs:892]: Is there potential for inlining code (e.g. here)?

Comment (michaelpj) [reply ghcide/src/Development/IDE/GHC/Compat/Core.hs]: yep, that one can go!

**Review by isovector — APPROVED** [review]
lgtm for tactics

**Review by konn — APPROVED** [review]
LGTM as for Splice.

**Pull request merged**